.github/workflows/docker-push-to-registries.yml

```diff
@@ -0,0 +1,137 @@
+          use APP_NAME as the application name.
+      APP_NAME:
+        required: true
+        type: string
+        description: |
+          The name for the application used as the main portion of the image name.
+      DOCKERFILE:
+        required: true
+        type: string
+        description: |
+          The name of the Dockerfile to build the image from.
+      PATH:
+        type: string
+        description: |
+          The path in which the Dockerfile is located. For current path leave empty. Otherwise eg. ./frontend
+        default: .
+      TARGET:
+        type: string
+        required: false
+        description: |
+          The build target within the Dockerfile to build the image from.
+      BUILD_ARGS:
+        type: string
+        required: false
+        description: |
+          Additional build arguments to pass to the Docker --build-arg flag.
+      VERSION:
+        required: true
+        type: string
+        description: |
+          The version for the image, used in tagging the image.
+      VERSION_SUFFIX:
+        required: false
+        type: string
+        description: |
+          Optional suffix for the version of the image, used in tagging the image.
+      IMAGE_SUFFIX:
+        required: false
+        type: string
+        description: |
+          Optional suffix used as part of the image name.
+      MAIN_BRANCH:
+        required: false
+        type: string
+        default: main
+        description: |
+          Specifies the main branch for the repository, defaults to main.  If this build is on the configured main
+          branch then the latest tag will be added to the image, if the build is on another branch then the development
+          tag will be added to the image.
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    permissions:
+      id-token: write
+      contents: read
+      packages: write
+    steps:
+      - name: Checkout
+        uses: actions/che ckout@v3
+
+      - name: Set up Docker buildx
+        id: buildx
+        uses: docker/setup-buildx-action@v2
+
+      - name: Enable buildkit cache
+        uses: actions/cache@v2
+        with:
+          path: /tmp/buildkit-cache/buildkit-state.tar
+          key: ${{ runner.os }}-buildkit-${{ github.sha }}
+          restore-keys: |
+            ${{ runner.os }}-buildkit-
+
+      - name: Load buildkit state from cache
+        uses: dashevo/gh-action-cache-buildkit-state@v1
+        with:
+          builder: buildx_buildkit_${{ steps.buildx.outputs.name }}0
+          cache-path: /tmp/buildkit-cache
+          cache-max-size: 3g
+
+      - name: Login to Docker Hub
+        uses: docker/login-action@v2
+        with:
+          username: ${{ secrets.DOCKER_USERNAME }}
+          password: ${{ secrets.DOCKER_PASSWORD }}
```

Comment:
Note this should only be used for images which will be public, not for private ones